core/capabilities/syncer.go

```diff
@@ -56,26 +71,486 @@ var defaultStreamConfig = p2ptypes.StreamConfig{
+			if myDONs[d.Id] {
+				myCapabilityDONs = append(myCapabilityDONs, d)
+			} else {
+				remoteCapabilityDONs = append(remoteCapabilityDONs, d)
+			}
+		}
+	}
+
+	// Now, if my node is a workflow DON, let's setup any shims
+	// to external capabilities.
+	if len(myWorkflowDONs) > 0 {
+		myDON := myWorkflowDONs[0]
+
+		// TODO: this is a bit nasty; figure out how best to handle this.
+		if len(myWorkflowDONs) > 1 {
+			s.lggr.Warn("node is part of more than one workflow DON; assigning first DON as caller")
+		}
+
+		for _, rcd := range remoteCapabilityDONs {
+			err := s.addRemoteCapabilities(ctx, myDON, rcd, readerState)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// Finally, if I'm a capability DON, let's enable external access
+	// to the capability.
+	if len(myCapabilityDONs) > 0 {
+		for _, mcd := range myCapabilityDONs {
+			err := s.enableExternalAccess(ctx, myID, mcd, readerState, remoteWorkflowDONs)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func signersFor(don kcr.CapabilityRegistryDONInfo, state state) ([][]byte, error) {
+	s := [][]byte{}
+	for _, nodeID := range don.NodeP2PIds {
+		node, ok := state.IDsToNodes[nodeID]
+		if !ok {
+			return nil, fmt.Errorf("could not find node for id %s", nodeID)
+		}
+
+		s = append(s, node.Signer[:])
+	}
+
+	return s, nil
+}
+
+func toDONInfo(don kcr.CapabilityRegistryDONInfo) *capabilities.DON {
+	peerIDs := []p2ptypes.PeerID{}
+	for _, p := range don.NodeP2PIds {
+		peerIDs = append(peerIDs, p)
+	}
+
+	return &capabilities.DON{
+		ID:      fmt.Sprint(don.Id),
+		Members: peerIDs,
+		F:       don.F,
+	}
+}
+
+func toCapabilityType(capabilityType uint8) capabilities.CapabilityType {
+	switch capabilityType {
+	case 0:
+		return capabilities.CapabilityTypeTrigger
+	case 1:
+		return capabilities.CapabilityTypeAction
+	case 2:
+		return capabilities.CapabilityTypeConsensus
+	case 3:
+		return capabilities.CapabilityTypeTarget
+	default:
+		// Not found
+		return capabilities.CapabilityType(-1)
+	}
+}
+
+func (s *registrySyncer) addRemoteCapabilities(ctx context.Context, myDON kcr.CapabilityRegistryDONInfo, remoteDON kcr.CapabilityRegistryDONInfo, state state) error {
+	for _, c := range remoteDON.CapabilityConfigurations {
+		capability, ok := state.IDsToCapabilities[c.CapabilityId]
+		if !ok {
+			return fmt.Errorf("could not find capability matching id %s", c.CapabilityId)
+		}
+
+		switch toCapabilityType(capability.CapabilityType) {
+		case capabilities.CapabilityTypeTrigger:
+			newTriggerFn := func(info capabilities.CapabilityInfo) (capabilityService, error) {
+				if !strings.HasPrefix(info.ID, "streams-trigger") {
+					return nil, errors.New("not supported: trigger capability does not have id = streams-trigger")
+				}
+
+				codec := streams.NewCodec(s.lggr)
+
+				signers, err := signersFor(remoteDON, state)
+				if err != nil {
+					return nil, err
+				}
+
+				aggregator := triggers.NewMercuryRemoteAggregator(
```

Comment:
```suggestion
				aggregator := triggers.NewRemoteAggregator(
```
Could we make this generic? I imagine each trigger could expose a `RemoteAggregator`

Reply:
No because it's not generic, intentionally so :)